### ISSUE TITLE
Strange behavior of UI yaml editors

### ISSUE THREAD

**Issue by VDRainer** [home-assistant/frontend#8496]
**Checklist**

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

**Describe the issue you are experiencing**

I see this only in Firefox, Chrome seems to work.
If i click into the yaml editor in Dev Tools/services or templates for the first time i see no cursor neither in light nor in dark mode.
Now, if i reload the browser window, i see a cursor, but when i want to mark multiple characters i see errors in home-assistant.log.
```
2021-03-01 09:38:30 ERROR (MainThread) [frontend.js.latest.202102260] http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:0 NS_ERROR_FAILURE: 
2021-03-01 09:38:30 ERROR (MainThread) [frontend.js.latest.202102260] http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:0 NS_ERROR_FAILURE: 
2021-03-01 09:38:30 ERROR (MainThread) [frontend.js.latest.202102260] http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:0 NS_ERROR_FAILURE: 
2021-03-01 09:38:30 ERROR (MainThread) [frontend.js.latest.202102260] http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:0 NS_ERROR_FAILURE: 
2021-03-01 09:38:30 ERROR (MainThread) [frontend.js.latest.202102260] http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:0 NS_ERROR_FAILURE: 
2021-03-01 09:38:31 ERROR (MainThread) [frontend.js.latest.202102260] http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:0 NS_ERROR_FAILURE: 
2021-03-01 09:38:31 ERROR (MainThread) [frontend.js.latest.202102260] http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:0 NS_ERROR_FAILURE:
```
Sometimes you have to change from templates to services to see this behavior.

**Describe the behavior you expected**

Easy edit yaml in services and templates (not sure if other UI editors are also affected, using only yaml mode).

**Steps to reproduce the issue**

1. Go to Dev Tools services -> yaml mode
2. click on the code, no corsor appears
3. Reload browser (CTRL-F5), cursor appears
4. Try to mask multiple characters, sometimes works, sometimes raises errors in log.

**What version of Home Assistant Core has the issue?**

core-2021.3.0b4

**What was the last working version of Home Assistant Core?**

core-2021.2.4

**In which browser are you experiencing the issue with?**

Firefox 86.0

**Which operating system are you using to run this browser?**

Ubuntu 18.03

**State of relevant entities**

```yaml
# Paste your state here.

```

**Problem-relevant frontend configuration**

```yaml
# Paste your YAML here.

```

**Javascript errors shown in your browser console/inspector**

```txt
Uncaught 
Exception { name: "NS_ERROR_FAILURE", message: "", result: 2147500037, filename: "http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js", lineNumber: 1, columnNumber: 0, data: null, stack: "updateSelection/<@http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:79596\nignore@http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:123040\nupdateSelection@http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:79291\nupdate@http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:77288\nupdate@http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:131846\nSi/this._dispatch<@http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:130426\ndispatch@http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:131240\nselect@http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:93963\nmove@http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:93520\nEventListener.handleEvent*De@http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:92678\nstartMouseSelection@http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:92331\n19753/Ee.mousedown@http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:96947\nTe/<@http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:90849\nEventListener.handleEvent*Te@http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:90641\nSi@http://192.168.77.67:8123/frontend_latest/chunk.83a9dc57827cc9362be1.js:1:130753\nvalue@http://192.168.77.67:8123/frontend_latest/chunk.f17c40c6007028fdab2f.js:1:10413\nasync*value@http://192.168.77.67:8123/frontend_latest/chunk.f17c40c6007028fdab2f.js:1:9977\nperformUpdate@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:111193\n_enqueueUpdate@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:110704\nasync*requestUpdateInternal@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:110494\ninitialize@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:108849\nC@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:106761\nr@http://192.168.77.67:8123/frontend_latest/chunk.f17c40c6007028fdab2f.js:1:8516\n_clone@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:123498\n__commitTemplateResult@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:120020\ncommit@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:119307\na@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:122700\n15652/T.render@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:114579\nupdate@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:114021\nperformUpdate@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:111052\n_enqueueUpdate@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:110704\nasync*requestUpdateInternal@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:110494\ninitialize@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:108849\ninitialize@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:113259\nC@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:106761\nT@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:112688\nF@http://192.168.77.67:8123/frontend_latest/chunk.d4810c9a130f714a7d63.js:736:7834\n_clone@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:123498\n__commitTemplateResult@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:120020\ncommit@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:119307\nupdate@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:123414\n__commitTemplateResult@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:119950\ncommit@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:119307\na@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:122700\n15652/T.render@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:114579\nupdate@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:114021\nperformUpdate@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:111052\n_enqueueUpdate@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:110704\nasync*requestUpdateInternal@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:110494\nrequestUpdate@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:110543\nset/<@http://192.168.77.67:8123/frontend_latest/chunk.822787cd796a0b79be33.js:1:1447\nset@http://192.168.77.67:8123/frontend_latest/chunk.822787cd796a0b79be33.js:1:1471\nvalue@http://192.168.77.67:8123/frontend_latest/chunk.1d02ff9964f9043f3ffd.js:272:1148\nhandleEvent@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:122285\ng/this.__boundHandleEvent@http://192.168.77.67:8123/frontend_latest/app.c6ae8c21.js:849:121652\n" }

```

**Comment by bramkragten** [comment]
This seems to be an upstream issue, found this old issue: <https://github.com/codemirror/codemirror.next/issues/141>

Seems that if you click out of the editor and then back in it does work.

**Comment by roblamoreaux** [comment]
I think I am seeing something related to this in 2021.3.0b3 through 2021.3.0b6. If I go in to edit a picture element the lines of YAML are incomplete. I get somewhere between 70 and 90 lines showing up, but I can scroll further through blank lines. Clicking on the editor section causes the text to suddenly appear. If I then scroll it is gone until I click again.

This is with the latest version of Edge and Chrome 88.0.4324.190 as well as the Android App.

**Comment by scooper1** [comment]
the problem is still present in beta 7   - problem with scroll  in  lovelace card code editor  - buffer is only partly loaded until you click in a blank area to show  some more text - happens after about 90 to 120 lines - variable both directions.

**Comment by bramkragten** [comment]
These are upstream issues, not much I can do about it. Please report them to codemirror.

**Comment by scooper1** [comment]
The chrome issue should I do a different Bug report to separate the issue Not sure what to report as  to what the issue for codemirror is as this only happens during the Lovelace card editor not the raw config editorOn 3 Mar 2021 09:56, Bram Kragten <notifications@github.com> wrote:
These are upstream issues, not much I can do about it. Please report them to codemirror.

—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.
On 3 Mar 2021 09:56, Bram Kragten <notifications@github.com> wrote:
These are upstream issues, not much I can do about it. Please report them to codemirror.

—You are receiving this because you commented.Reply to this email directly, view it on GitHub, or unsubscribe.
On 3 Mar 2021 09:56, Bram Kragten <notifications@github.com> wrote:
These are upstream issues, not much I can do about it. Please report them to codemirror.

**Comment by KennethLavrsen** [comment]
I see that my just reported https://github.com/home-assistant/frontend/issues/8534 is part of this issue

This is not just Firefox. It is also Google Chrome and Safari on IOS. You cannot edit YAML anymore. It worked just until we upgraded só if this is an updstream issue then the cure is to revert whatever upstream dependency back to previous version. Right now you cannot edit anything beyond the most simple in Lovelace.

**Comment by roblamoreaux** [comment]
It looks like this was reported to the CodeMirror repository back in November but wasn't really addressed. 
https://github.com/codemirror/CodeMirror/issues/6483

**Comment by bramkragten** [comment]
That is the repository for the old version of CodeMirror, not the new one:
https://github.com/codemirror/codemirror.next/issues

**Comment by roblamoreaux** [comment]
Shows how little I know about the underlying workings. I'm just a HW guy trying to provide feedback.

**Comment by VDRainer** [comment]
Fixed in 2021.3.3 :smile:
Thanks!
